File: person_task_manager/server/src/kernel/util/build-headers.ts
import { config } from "../config/security-configuration";
import { encrypt } from "../../infrastructure/security/encrypt-rsa";

export const buildDefaultHeaders = (headers: Record<string, string>) => {
    return {
        'Content-Type': 'application/json',
        'Accept': 'application/json',
        ...headers
    };
}

export const buildAuthorizationHeaders = (service: string, userId: number, headers: Record<string, string>) => {
    return {
        'Content-Type': 'application/json',
        'Accept': 'application/json',
        'Service-Token': encrypt(`${service}::${config.privateToken}::${userId}`),
        'Service': service,
        ...headers
    }
}

File: middleware_loader/kernel/utils/build_headers.go
package utils

import (
	"fmt"
	"middleware_loader/infrastructure/security"
	"middleware_loader/kernel/configs"
)

var config = configs.SecurityConfig{}
var securityConfig, _ = config.LoadSecurityEnv()
var privateToken = securityConfig.PrivateToken

func BuildDefaultHeaders() map[string]string {
	headers := make(map[string]string)
	headers["Content-Type"] = "application/json"
	headers["Accept"] = "application/json"
	return headers
}

func BuildAuthorizationHeaders(service string, userId string) map[string]string {
	headers := BuildDefaultHeaders()
	headers["Service"] = service
	token, err := security.Encrypt(service + "::" + privateToken + "::" + userId)
	if err != nil {
		fmt.Println("Error encrypting token: ", err)
		return headers
	}
	headers["Service-Token"] = token
	return headers
}